pkg/util/oidcbuilder/jwks.go

```diff
@@ -0,0 +1,136 @@
+package oidcbuilder
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"gopkg.in/go-jose/go-jose.v2"
+)
+
+func CreateKeyPair() (encPrivateKey []byte, encPublicKey []byte, err error) {
+	bitSize := 4096
+
+	// Generate RSA keypair
+	privateKey, err := rsa.GenerateKey(rand.Reader, bitSize)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to generate private key")
+	}
+	encodedPrivateKey := pem.EncodeToMemory(&pem.Block{
+		Type:    "RSA PRIVATE KEY",
+		Headers: nil,
+		Bytes:   x509.MarshalPKCS1PrivateKey(privateKey),
+	})
+
+	// Generate public key from private keypair
+	pubKeyBytes, err := x509.MarshalPKIXPublicKey(&privateKey.PublicKey)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to generate public key from private")
```

Comment:
```suggestion
		return nil, nil, errors.Wrapf(err, "failed to serialize public key")
```

See my comment a few lines above this for the reasoning behind this suggestion.